src/aa_remove_data/remove_data.py

```diff
@@ -0,0 +1,486 @@
+
+
+def aa_remove_every_nth():
+    parser = argparse.ArgumentParser()
+    parser = add_generic_args(parser)
+    parser.add_argument("n", type=int, help="remove every nth data point")
+    args = parser.parse_args()
+
+    if args.new_filename is None:
+        new_pb = Path(args.filename)
+    else:
+        new_pb = Path(args.new_filename)
+
+    if args.backup_filename is None:
+        backup_pb = Path(args.filename.strip(".pb") + "_backup.pb")
+    else:
+        backup_pb = Path(args.backup_filename)
+
+    validate_pb_file(args.filename)
+    validate_pb_file(new_pb)
+    validate_pb_file(backup_pb)
+
+    txt_filepath = new_pb.with_suffix(".txt")
+    pb = PBUtils(chunk_size=args.chunk)
+    initial = 0
+    while pb.read_done is False:
+        pb.read_pb(Path(args.filename))
+        pb.write_pb(backup_pb)
+        pb.samples = remove_every_nth(
+            pb.samples, args.n, block_size=args.block, initial=initial
+        )
+        pb.write_pb(new_pb)
+        if args.write_txt:
+            txt_filepath = Path(str(new_pb).strip(".pb") + ".txt")
+            pb.write_to_txt(txt_filepath)
+        initial = (args.chunk + initial) % (args.n * args.block)
+
+
+def aa_remove_data_before():
+    parser = argparse.ArgumentParser()
+    parser = add_generic_args(parser)
+    parser.add_argument(
+        "--ts",
+        nargs="+",
+        type=int,
+        required=True,
+        metavar="timestamp",
+        help="timestamp in the form 'month day hour minute second nanosecond' "
+        + "- month is required (default: {month} 1 0 0 0 0)",
+    )
+    args = parser.parse_args()
+
+    if args.new_filename is None:
+        new_pb = Path(args.filename)
+    else:
+        new_pb = Path(args.new_filename)
+
+    if args.backup_filename is None:
+        backup_pb = Path(args.filename.strip(".pb") + "_backup.pb")
+    else:
+        backup_pb = Path(args.backup_filename)
+
+    validate_pb_file(args.filename)
+    validate_pb_file(new_pb)
+    validate_pb_file(backup_pb)
+
+    pb_header = PBUtils(Path(args.filename), chunk_size=0)
+    year = pb_header.header.year
+    timestamp = args.ts
+    assert len(timestamp) <= 6, (
+        "Give timestamp in the form 'month.day.hour.minute.second.nanosecond'. "
+        + "Month is required. All must be integers."
+    )
+    if len(timestamp) == 6:
+        nano = timestamp.pop(5)
+    else:
+        nano = 0
+    if len(timestamp) == 1:
+        timestamp.append(1)
+
+    diff = datetime(*([year] + timestamp)) - datetime(year, 1, 1)
+    seconds = int(diff.total_seconds())
+    pb = PBUtils(chunk_size=args.chunk)
+    while pb.read_done is False:
+        pb.read_pb(Path(args.filename))
+        pb.write_pb(backup_pb)
+        pb.samples = remove_before_ts(pb.samples, seconds, nano=nano)
+        pb.write_pb(new_pb)
+        if args.write_txt:
+            txt_filepath = Path(str(new_pb).strip(".pb") + ".txt")
+            pb.write_to_txt(txt_filepath)
+
+
+def aa_remove_data_after():
```

Comment:
These entry points have no docstrings. Could you add something similar to what you've done in pb_utils.py? A single line statement should be enough